server/src/main/scala/org/http4s/server/middleware/authentication/DigestAuth.scala

```diff
@@ -35,11 +35,22 @@ import scala.concurrent.duration._
   */
 object DigestAuth {
 
+  type PlainAuthenticationStore[F[_], A] = String => F[Option[(A, String)]]
+  sealed trait AuthenticationStore[F[_], A]
```

Comment:
A `type` alias has no binary signature, so promoting to a `trait` is binary-compatible. However, it would be source-breaking ... but maybe that's okay, since the benefit is tangible.

I'm not sure if we need the `PlainAuthenticationStore` alias, we can just use `String => F[Option[(A, String)]]` and deprecate anything using that anyway.

Reply:
Yeah, I think the self-documenting nature of the compiler error is worth the source compat breakage. I was thinking of doing something complex like

```scala
sealed class AuthenticationStore[F[_], A] extends Function1[String, F[Option[(A, String)]]] { ... }
```

in order to maintain some sort of compatibility, but it just made everything way too complicated for negligible benefit.

Reply:
Could we call the new one `AuthStore` instead, and deprecate the `AuthenticationStore` alias?